soap/index.d.ts

```diff
@@ -19,12 +19,22 @@ export class ClientSSLSecurity implements Security {
     constructor(key: Buffer | string, cert: Buffer | string, defaults?: any);
 }
 
-interface Client extends events.EventEmitter {
-    setSecurity(s: Security): void;
-    [method: string]: (args: any, fn: (err: any, result: any) => void, options?: any, extraHeaders?: any) => void;
-    addSoapHeader(headJSON: any): void;
-    setEndpoint(endpoint: string): void;
+export interface Client extends events.EventEmitter {
+    [method: string]: any;
```

Comment:
I re-added the old type and moved it to the bottom of the definitions, but it is now throwing the error I mentioned before for two of the functions.